File: golang/73_set_matrix_zeroes.go
func setZeroes(matrix [][]int) {
	row0, col0 := false, false
	m := len(matrix)
	n := len(matrix[0])
	for i := 0; i < m; i++ {
		if matrix[i][0] == 0 {
			col0 = true
		}
	}
	for i := 0; i < n; i++ {
		if matrix[0][i] == 0 {
			row0 = true
		}
	}
	for i := 1; i < m; i++ {
		for j := 1; j < n; j++ {
			if matrix[i][j] == 0 {
				matrix[i][0] = 0
				matrix[0][j] = 0
			}
		}
	}
	for i := 1; i < m; i++ {
		for j := 1; j < n; j++ {
			if matrix[i][0] == 0 || matrix[0][j] == 0 {
				matrix[i][j] = 0
			}
		}
	}
	if row0 == true {
		for i := 0; i < n; i++ {
			matrix[0][i] = 0
		}
	}
	if col0 == true {
		for i := 0; i < m; i++ {
			matrix[i][0] = 0
		}
	}
}

File: python/73_set_matrix_zeroes.py
from typing import List

class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        row_0 = col_0 = False
        m = len(matrix)
        n = len(matrix[0])
        for i in range(0, m):
            if matrix[i][0] == 0:
                col_0 = True
        for i in range(0, n):
            if matrix[0][i] == 0:
                row_0 = True
        for i in range(1, m):
            for j in range(1, n):
                if matrix[i][j] == 0:
                    matrix[i][0] = 0
                    matrix[0][j] = 0
        for i in range(1, m):
            for j in range(1, n):
                if matrix[i][0] == 0 or matrix[0][j] == 0:
                    matrix[i][j] = 0
        if row_0:
            for i in range(0, n):
                matrix[0][i] = 0
        if col_0:
            for i in range(0, m):
                matrix[i][0] = 0
        